src/CryptoNoteCore/RocksDBWrapper.cpp

```diff
@@ -52,6 +52,12 @@ void RocksDBWrapper::init(const DataBaseConfig& config) {
   rocksdb::DB* dbPtr;
 
   rocksdb::Options dbOptions = getDBOptions(config);
+  if (config.getCompressionEnabled()) {
+      dbOptions.compression_opts.strategy = 9; // level 9

```

Comment:
What we missed is .level :)

Reply:
Thanks. I think I missed also the dict part at column setting below. Settings here won't affect.

Reply:
```
Process ./data/DB/000027.sst
Sst file format: block-based
Table Properties:
------------------------------
  # data blocks: 16803
  # entries: 296259
  # deletions: 0
  # merge operands: 0
  # range deletions: 0
  raw key size: 12739137
  raw average key size: 43.000000
  raw value size: 121066608
  raw average value size: 408.651241
  data block size: 107378955
  index block size (user-key? 0, delta-value? 0): 915999
  filter block size: 0
  (estimated) table size: 108294954
  filter policy name: N/A
  prefix extractor name: nullptr
  column family ID: 0
  column family name: default
  comparator name: leveldb.BytewiseComparator
  merge operator name: nullptr
  property collectors names: []
  SST file compression algo: ZSTD
  SST file compression options: window_bits=-14; level=16; strategy=0; max_dict_bytes=16384; zstd_max_train_bytes=262144; enabled=1;
  creation time: 1567296028
  time stamp of earliest key: 0
  file creation time: 1567296054
  Raw user collected properties
------------------------------
  # rocksdb.block.based.table.index.type: 0x00000000
  # rocksdb.block.based.table.prefix.filtering: 0x30
  # rocksdb.block.based.table.whole.key.filtering: 0x31
  # rocksdb.deleted.keys: 0x00
  # rocksdb.merge.operands: 0x00
```
One output from sst_dump. I adjust the `max_dict_bytes` and `zstd_max_train_bytes`. Still doing some process of testings :)

Reply:
I think I misconfigure dict with this commit.
It should be :
    // zstd compression settings
    if (config.getCompressionEnabled()) {
        fOptions.compression_opts.level = 9; // level 9
        fOptions.compression_opts.strategy = 9; // level 9
        fOptions.compression_opts.max_dict_bytes = 1 << 16;        // 64KB
        fOptions.compression_opts.zstd_max_train_bytes = 1 << 20;  // 1024KB
        fOptions.compression_opts.enabled = true;
    }

I'm testing it with wrkz

Reply:
Ouff, bad news. With new dictionary setting, compression ratio is much worse.